corehq/apps/domain/deletion.py

```diff
@@ -164,22 +164,24 @@ def _terminate_subscriptions(domain_name):
         ).update(is_hidden_to_ops=True)
 
 
-def _delete_all_cases(domain_name):
+def delete_all_cases(domain_name):
     logger.info('Deleting cases...')
     case_ids = CommCareCase.objects.get_case_ids_in_domain(domain_name)
+    case_ids += CommCareCase.objects.get_deleted_case_ids_in_domain(domain_name)
```

Comment:
In the case of the `delete_domain` management command, previously this was a no-op because all cases had already been deleted. Now this will load all case ids into memory in a single query. What do you think of changing this to use `case_ids = iter_ids(CommCareCase, 'case_id', domain_name)` as the management command did? That would be a much more efficient way of deleting cases without loading all case ids in one single batch.

`iter_ids` (defined in the management command module) should be changed to not use `with_progress_bar` since this may run in contexts where calling `print()` is inappropriate.

Reply:
That makes sense to me. I moved `iter_ids` to `deletion.py` since it was no longer being referenced in the delete_domain command directly, and seemed like a weird place for it to live now. See 64287d0012619cdd77f9138b2027814a52dc3e2e.